cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractKafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -0,0 +1,230 @@
+import io.strimzi.operator.cluster.operator.resource.cruisecontrol.MockCruiseControl;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.operator.MockCertManager;
+import io.strimzi.platform.KubernetesVersion;
+import io.strimzi.test.ReadWriteUtils;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.mockkube3.MockKube3;
+import io.vertx.core.Vertx;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(VertxExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractKafkaRebalanceAssemblyOperatorTest {
+    protected static final String HOST = "localhost";
+    protected static final String RESOURCE_NAME = "my-rebalance";
+    protected static final String CLUSTER_NAME = "kafka-cruise-control-test-cluster";
+    protected static final Kafka KAFKA = new KafkaBuilder()
+            .withNewMetadata()
+            .withName(CLUSTER_NAME)
+            .withAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled", Annotations.ANNO_STRIMZI_IO_KRAFT, "enabled"))
+            .endMetadata()
+            .withNewSpec()
+            .withNewKafka()
+            .withListeners(new GenericKafkaListenerBuilder()
+                    .withName("plain")
+                    .withPort(9092)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(false)
+                    .build())
+            .endKafka()
+            .withNewCruiseControl()
+            .endCruiseControl()
+            .endSpec()
+            .build();
+    protected static final KafkaRebalanceSpec EMPTY_KAFKA_REBALANCE_SPEC = new KafkaRebalanceSpecBuilder().build();
+    protected static final PlatformFeaturesAvailability PFA = new PlatformFeaturesAvailability(false, KubernetesVersion.MINIMAL_SUPPORTED_VERSION);
+    protected static KubernetesClient client;
+    protected static MockKube3 mockKube;
+    protected static Vertx vertx;
+    protected static WorkerExecutor sharedWorkerExecutor;
+    protected String namespace;
+    protected KafkaRebalanceAssemblyOperator krao;
+    protected ResourceOperatorSupplier supplier;
+
+    protected static int cruiseControlPort;
+    protected static File tlsKeyFile;
+    protected static File tlsCrtFile;
+    protected static MockCruiseControl cruiseControlServer;
```

Comment:
Why not put these fields with the other `protected static` fields above?

Reply:
https://github.com/strimzi/strimzi-kafka-operator/pull/11348#discussion_r2089293297